detox/e2e/test/custom_statuses/custom_status_expiry.e2e.js

```diff
@@ -0,0 +1,214 @@
+        await ChannelScreen.postMessage(message);
+
+        // # Open user profile screen
+        await openSettingsSidebar();
+        await UserProfileScreen.open();
+        await UserProfileScreen.toBeVisible();
+
+        // * Check if custom status is present in the user profile screen and close it
+        await expect(element(by.id(`custom_status.emoji.${defaultStatus.emoji}`)).atIndex(0)).toExist();
+        await expect(element(by.text(defaultStatus.text).withAncestor(by.id(UserProfileScreen.testID.customStatus)))).toBeVisible();
+        await expect(element(by.text('STATUS (UNTIL ' + expiryText + ')'))).toBeVisible();
+
+        await UserProfileScreen.close();
+
+        // # Open the main sidebar and click on more direct messages button
+        await ChannelScreen.openMainSidebar();
+        await MoreDirectMessagesScreen.open();
+
+        // # Type the logged in user's username and tap it to open the DM
+        await MoreDirectMessagesScreen.searchInput.typeText(testUser.username);
+        await MoreDirectMessagesScreen.getUserAtIndex(0).tap();
+
+        // # Open the channel info screen
+        await ChannelInfoScreen.open();
+
+        // * Check if the custom status is present in the channel info screen and then close the screen
+        await expect(element(by.id(`custom_status.emoji.${defaultStatus.emoji}`)).atIndex(0)).toExist();
+        await expect(element(by.text(defaultStatus.text).withAncestor(by.id(ChannelInfoScreen.testID.headerCustomStatus)))).toBeVisible();
+        await expect(element(by.text('Until ' + expiryText))).toBeVisible();
+
+        await ChannelInfoScreen.close();
+    });
+
+    it("MM-T4092 RN apps: Custom Status Expiry - Editing 'Clear After' Time (mobile)", async () => {
+        // # Open custom status screen
+        await openSettingsSidebar();
+        await CustomStatusScreen.open();
+
+        // * Check if all the default suggestions are visible
+        const isSuggestionPresentPromiseArray = [];
+        defaultCustomStatuses.map(async (text) => {
+            isSuggestionPresentPromiseArray.push(expect(getCustomStatusSuggestion(text)).toBeVisible());
+        });
+        await Promise.all(isSuggestionPresentPromiseArray);
+
+        // * Tap a suggestion and check if it is selected
+        await tapSuggestion(defaultStatus);
+        await expect(element(by.id(`custom_status.duration.${defaultStatus.duration}`))).toBeVisible();
+
+        // # Click on the Clear After option and check if the modal opens
+        await ClearAfterScreen.open();
+
+        // # Select a different expiry time and check if it is shown in Clear After
+        await ClearAfterScreen.tapSuggestion('four_hours');
+        await ClearAfterScreen.close();
+
+        // Check if selected duration is shown in clear after
+        await expect(element(by.id('custom_status.duration.four_hours'))).toBeVisible();
```

Comment:
I'm fine to leave it as is but testID should be declared in the corresponding test screen/component. In the case of dynamically generated testIDs, the test screen/component should expose a convenient function for such purpose. The idea is not to hard-code in multiple places.